pkg/pluginmanager/manager.go

```diff
@@ -989,7 +996,42 @@ func SyncPlugins() error {
 	if err != nil {
 		errList = append(errList, err)
 	}
-	if installedPlugins, err := pluginsupplier.GetInstalledServerPlugins(); err == nil {
+	err = installDiscoveredContextPlugins(plugins)
+	if err != nil {
+		errList = append(errList, err)
+	}
+	return kerrors.NewAggregate(errList)
+}
+
+// SyncPluginsForContextType installs the plugins for given ContextType
+func SyncPluginsForContextType(contextType configtypes.ContextType) error {
+	currentContextMap, err := configlib.GetAllActiveContextsMap()
```

Comment:
Instead of looping through the map, I think you can directly call `config.GetActiveContext(..)`

Reply:
thanks, updated.